libbeat/dashboards/import_dashboards.go

```diff
@@ -0,0 +1,559 @@
+	cl.flagSet.StringVar(&cl.opt.Index, "i", "", "Overwrites the Elasticsearch index name. For example you can replaces metricbeat-* with custombeat-*")
+	cl.flagSet.StringVar(&cl.opt.Dir, "dir", "", "Directory containing the subdirectories: dashboard, visualization, search, index-pattern. eg. kibana/")
+	cl.flagSet.StringVar(&cl.opt.File, "file", "", "Zip archive file containing the Beats dashboards.")
+	cl.flagSet.StringVar(&cl.opt.Url, "url", "", "URL to the zip archive containing the Beats dashboards")
+
+	return &cl, nil
+}
+
+func (cl *CommandLine) Read() error {
+
+	if err := cl.flagSet.Parse(os.Args[1:]); err != nil {
+		return err
+	}
+
+	if cl.opt.Url == "" && cl.opt.File == "" && cl.opt.Dir == "" {
+		return errors.New("ERROR: Missing input. Please specify one of the options -file, -url or -dir")
+	}
+
+	return nil
+}
+
+func New() (*Importer, error) {
+	importer := Importer{}
+
+	/* command line */
+	cl, err := ParseCommandLine()
+	if err != nil {
+		cl.flagSet.Usage()
+		return nil, err
+	}
+	err = cl.Read()
+	if err != nil {
+		return nil, err
+	}
+	importer.cl = cl
+
+	/* connect to Elasticsearch */
+	fmtstr, err := fmtstr.CompileEvent(cl.opt.Index)
+	if err != nil {
+		return nil, fmt.Errorf("fail to compile the Elasticsearch index pattern: %s", err)
+	}
+	indexSel := outil.MakeSelector(outil.FmtSelectorExpr(fmtstr, ""))
+	client, err := elasticsearch.NewClient(
+		elasticsearch.ClientSettings{
+			URL:      cl.opt.ES,
+			Index:    indexSel,
+			Username: cl.opt.User,
+			Password: cl.opt.Pass,
+			Timeout:  60 * time.Second,
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("fail to connect to Elasticsearch: %s", err)
+	}
+	importer.client = client
+
+	return &importer, nil
+
+}
+
+func (imp Importer) CreateIndex() error {
+	imp.client.CreateIndex(imp.cl.opt.KibanaIndex, nil)
+	_, _, err := imp.client.CreateIndex(imp.cl.opt.KibanaIndex+"/_mapping/search",
+		common.MapStr{
+			"search": common.MapStr{
+				"properties": common.MapStr{
+					"hits": common.MapStr{
+						"type": "integer",
+					},
+					"version": common.MapStr{
+						"type": "integer",
+					},
+				},
+			},
+		})
+	if err != nil {
+		fmt.Printf("fail to set the mapping. Error: %s\n", err)
+	}
+	return nil
+}
+
+func (imp Importer) ImportJsonFile(_type string, file string) error {
+
+	path := "/" + imp.cl.opt.KibanaIndex + "/" + _type
+
+	reader, err := ioutil.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("fail to read %s. Error: %s", file, err)
+	}
+	var json_content map[string]interface{}
+	json.Unmarshal(reader, &json_content)
+	fileBase := strings.TrimSuffix(filepath.Base(file), filepath.Ext(file))
+
+	err = imp.client.LoadJson(path+"/"+fileBase, json_content)
+	if err != nil {
+		return fmt.Errorf("fail to load %s under %s/%s: %s", file, path, fileBase, err)
+	}
+
+	return nil
+}
+
+func (imp Importer) ImportDashboard(file string) error {
+
+	fmt.Println("Import dashboard ", file)
+
+	/* load dashboard */
```

Comment:
This is how I called the scripts first, `load_dashboards`, but based on the feedback I got, I decided to rename it to `import_dashboards`, as people had troubles understanding `load`, especially for someone that it's used with the Kibana terminology.